v2/dtos/requests/deviceservice.go

```diff
@@ -60,3 +62,121 @@ func AddDeviceServiceReqToDeviceServiceModels(addRequests []AddDeviceServiceRequ
+// This object and its properties correspond to the UpdateDeviceServiceRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/UpdateDeviceServiceRequest
+type UpdateDeviceServiceRequest struct {
+	common.BaseRequest `json:",inline"`
+	Service            dtos.UpdateDeviceServiceDTO `json:"service"`
+}
+
+// Validate satisfies the Validator interface
+func (ds UpdateDeviceServiceRequest) Validate() error {
+	err := v2.Validate(ds)
+	return err
+}
+
+// UnmarshalJSON implements the Unmarshaler interface for the UpdateDeviceServiceRequest type
+func (ds *UpdateDeviceServiceRequest) UnmarshalJSON(b []byte) error {
+	var alias struct {
+		common.BaseRequest
+		Service dtos.UpdateDeviceServiceDTO
+	}
+	if err := json.Unmarshal(b, &alias); err != nil {
+		return v2.NewErrContractInvalid("Failed to unmarshal request body as JSON.")
+	}
+
+	*ds = UpdateDeviceServiceRequest(alias)
+
+	// validate AddDeviceServiceRequest DTO
+	if err := ds.Validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ReplaceDeviceServiceModelFieldsWithDTO replace model's fields with DTO patch
+func ReplaceDeviceServiceModelFieldsWithDTO(ds *models.DeviceService, patch dtos.UpdateDeviceServiceDTO) {
+	if patch.Id != nil && patch.DeviceServiceName != nil {
+		ds.Name = *patch.DeviceServiceName
+	}
+	if patch.OperatingState != nil {
+		ds.OperatingState = models.OperatingState(*patch.OperatingState)
+	}
+	if patch.AdminState != nil {
+		ds.AdminState = models.AdminState(*patch.AdminState)
+	}
+	if patch.Labels != nil {
+		ds.Labels = patch.Labels
+	}
+	if patch.BaseAddress != nil {
+		ds.BaseAddress = *patch.BaseAddress
+	}
+}
+
+// TODO how to validate the map
+type UpdateDeviceServiceRequestMap struct {
+	common.BaseRequest `json:",inline"`
+	Service            map[string]interface{} `json:"service"`
+}
+
+// ReplaceDeviceServiceModelFieldsWithMap replace model's fields with map patch
+func ReplaceDeviceServiceModelFieldsWithMap(ds *models.DeviceService, patch map[string]interface{}) {
+	//_, idExist := patch["id"]
+	//deviceServiceName, deviceServiceNameExist := patch["deviceServiceName"]
+	//if idExist && deviceServiceNameExist {
+	//	v, ok := deviceServiceName.(string)
+	//	if ok {
+	//		ds.Name = v
+	//	}
+	//}
+	//
+	//if operatingState, exist := patch["operatingState"]; exist {
+	//	v, ok := operatingState.(string)
+	//	if ok {
+	//		ds.OperatingState = models.OperatingState(v)
+	//	}
+	//}
+	//if adminState, exist := patch["adminState"]; exist {
+	//	v, ok := adminState.(string)
+	//	if ok {
+	//		ds.AdminState = models.AdminState(v)
+	//	}
+	//}
+	//
+	//if labels, exist := patch["labels"]; exist {
+	//	v, ok := labels.([]string)
+	//	if ok {
+	//		ds.Labels = v
+	//	}
+	//}
+	//if baseAddress, exist := patch["baseAddress"]; exist {
+	//	v, ok := baseAddress.(string)
+	//	if ok {
+	//		ds.BaseAddress = v
+	//	}
+	//}
+
+	vals := reflect.ValueOf(ds)
```

Comment:
I agree the the other approach with pointers is cleaner and much easier to follow.

Reply:
Thanks, I will use the pointer in implementation and update the swagger file with the issue https://github.com/edgexfoundry/edgex-go/issues/2639.